macosx/DefaultAppHelper.mm

```diff
@@ -0,0 +1,171 @@
+// This file Copyright © 2007-2023 Transmission authors and contributors.
+// It may be used under the MIT (SPDX: MIT) license.
+// License text can be found in the licenses/ folder.
+
+#import "DefaultAppHelper.h"
+
+#import <AppKit/AppKit.h>
+#import <UniformTypeIdentifiers/UniformTypeIdentifiers.h>
+
+static NSString* const kMagnetURLScheme = @"magnet";
+static NSString* const kTorrentFileType = @"org.bittorrent.torrent";
+
+UTType* GetTorrentFileType(void) API_AVAILABLE(macos(11.0))
+{
+    static UTType* result = nil;
+
+    static dispatch_once_t once;
+    dispatch_once(&once, ^{
+        result = [UTType exportedTypeWithIdentifier:kTorrentFileType conformingToType:UTTypeData];
+    });
+
+    return result;
+}
+
+@interface DefaultAppHelper ()
+
+@property(nonatomic, readonly) NSString* bundleIdentifier;
+
+@end
+
+@implementation DefaultAppHelper
+
+- (instancetype)init
+{
+    if (self = [super init])
+    {
+        _bundleIdentifier = NSBundle.mainBundle.bundleIdentifier;
+    }
+    return self;
+}
+
+- (BOOL)isDefaultForTorrentFiles
+{
+    if (@available(macOS 12, *))
+    {
+        UTType* file_type = GetTorrentFileType();
```

Comment:
Are we using snake case now?

Reply:
Fixed, I will look into actually checking naming conventions in ObjC++ with clang-format config.